### PR TITLE
mgr/dashboard: Add details to the modal which displays the `safe-to-d…

### PR DESCRIPTION
**Pull request by torchiaf**
…estroy` result

- Add warnings type information in the case of the OSDs are not safe to destroy
- Add info type information in the case of the OSDs are safe to destroy

Fixes: https://tracker.ceph.com/issues/37327
Signed-off-by: Francesco Torchia <francesco.torchia@suse.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by torchiaf** [comment]
Below, a demo showing all the possible warnings messages for different OSDs:

image

**Comment by torchiaf** [comment]
Below, the info message in case of the destroy action is safe:

image

**Comment by torchiaf** [comment]
The CI steps which are failed are not apparently related to the changes included in this PR.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.scss [hunk 1]
```diff
@@ -0,0 +1,8 @@
+.warnings-list {
+  margin-bottom: 0;
+  padding-left: 1.5em;
+}
+
+.danger {
+  margin-bottom: 1rem;
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.scss:7]: Wouldn't it be better to use `class="danger mb-X"` instead?

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.scss [hunk 2]
```diff
@@ -0,0 +1,8 @@
+.warnings-list {
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.scss:1]: Wouldn't it be better to use `class="warnings-list mb-0 pl-X"` instead?

**Review by votdev — CHANGES_REQUESTED** [review]

**Comment by votdev** [comment]
jenkins retest this please

**Review by votdev — APPROVED** [review]

**Comment by votdev** [comment]
jenkins retest this please

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.html [hunk 3]
```diff
@@ -64,13 +64,50 @@
...
              let-actionDescription="actionDescription"
              let-osdIds="osdIds">
   <div *ngIf="!safeToPerform"
-       class="danger">
-    <cd-alert-panel type="warning"
-                    i18n>The {selection.hasSingleSelection, select, true {OSD is} other {OSDs are}} not safe to be
-      {{ actionDescription }}! {{ message }}</cd-alert-panel>
+       class="danger mb-3">
+    <cd-alert-panel type="warning">
+      <ng-container i18n>
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.html:75]: [ng-container](https://angular.io/api/core/ng-container) is mostly used to hold the structural directives, but here a simple span will suffice, right?

**Review by nizamial09 — APPROVED** [review]
Thank you! @torchiaf LGTM! just a minor ask.

**Comment by nizamial09** [comment]
Thanks @torchiaf ! I added the backport for pacific and quincy

**Pull request merged**